src/base.ml

```diff
@@ -50,101 +48,132 @@ let render_texture (r : Sdl.renderer) (texture : Sdl.texture) (s : Screen.t) (bi
   Sdl.render_copy ~dst:dst r texture >|= fun () ->
   Sdl.render_present r
 
-(* ----- *)
-
-let run (title : string) (boot : boot_func option) (tick : tick_func) (s : Screen.t) =
-  let make_full = Array.to_list Sys.argv |> List.exists (fun a -> (String.compare a "-f") == 0) in
-
-  let s = match make_full with
-  | false -> s
+(* Poll SDL events and build the unified event queue.
+   Mouse events are handled by PlatformMouse.handle_event, which returns
+   an updated mouse state along with a list of unified events. *)
+let rec poll_all_events keys mouse acc =
```

Comment:
Can we add some tests to this similar the ones in test_mousesdl.ml? Just something that adds in a few events and makes sure they come out in the right order etc.